src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -487,16 +533,11 @@ protected virtual void CheckProjectCanBeRemoved(ProjectId projectId)
         {
         }
 
-        private void HandleProjectChange(ProjectId projectId, Func<Solution, Solution> getSolutionWithChangedProject)
+        private void HandleProjectChange(ProjectId projectId, Func<Solution, Solution> solutionTransformation)
         {
-            using (_serializationLock.DisposableWait())
```

Comment:
So should everything else still using the _serializationLock also just move to this? I'm now trying to grok how these different systems are supposed to interact...